src/modm/platform/adc/stm32f3/adc.hpp.in

```diff
@@ -199,6 +225,227 @@ public:
+		ExtTrigger19	= 0b10011,
+		ExtTrigger20	= 0b10100,
+		ExtTrigger21	= 0b10101,
+		ExtTrigger22	= 0b10110,
+		ExtTrigger23	= 0b10111,
+		ExtTrigger24	= 0b11000,
+		ExtTrigger25	= 0b11001,
+		ExtTrigger26	= 0b11010,
+		ExtTrigger27	= 0b11011,
+		ExtTrigger28	= 0b11100,
+		ExtTrigger29	= 0b11101,
+		ExtTrigger30	= 0b11110,
+		ExtTrigger31	= 0b11111,
+%% endif
+	};
+
+	enum class TriggerSourceInjected : uint8_t
+	{
+%% if target["family"] in ["g4"]
+%% if id in [1, 2]
+		Tim1Trgo		= 0b00000,
+		Tim1Cc4			= 0b00001,
+		Tim2Trgo		= 0b00010,
+		Tim2Cc1			= 0b00011,
+		Tim3Cc4			= 0b00100,
+		Tim4Trgo		= 0b00101,
+		ExtiLine15		= 0b00110,
+		Tim8Cc4			= 0b00111,
+		Tim1Trgo2		= 0b01000,
+		Tim8Trgo		= 0b01001,
+		Tim8Trgo2		= 0b01010,
+		Tim3Cc3			= 0b01011,
+		Tim3Trgo		= 0b01100,
+		Tim3Cc1			= 0b01101,
+		Tim6Trgo		= 0b01110,
+		Tim15Trgo		= 0b01111,
+		Tim20Trgo		= 0b10000,
+		Tim20Trgo2		= 0b10001,
+		Tim20Cc4		= 0b10010,
+		HrtimAdcTrg2	= 0b10011,
+		HrtimAdcTrg4	= 0b10100,
+		HrtimAdcTrg5	= 0b10101,
+		HrtimAdcTrg6	= 0b10110,
+		HrtimAdcTrg7	= 0b10111,
+		HrtimAdcTrg8	= 0b11000,
+		HrtimAdcTrg9	= 0b11001,
+		HrtimAdcTrg10	= 0b11010,
+		Tim16Cc1		= 0b11011,
+		//Reserved		= 0b11100,
+		Lptimout		= 0b11101,
+		Tim7Trgo		= 0b11110,
+		//Reserved		= 0b11111,
+%% elif id in [3, 4, 5]
+		Tim1Trgo		= 0b00000,
+		Tim1Cc4			= 0b00001,
+		Tim2Trgo		= 0b00010,
+		Tim8Cc2			= 0b00011,
+		Tim4Cc3			= 0b00100,
+		Tim4Trgo		= 0b00101,
+		Tim4Cc4			= 0b00110,
+		Tim8Cc4			= 0b00111,
+		Tim1Trgo2		= 0b01000,
+		Tim8Trgo		= 0b01001,
+		Tim8Trgo2		= 0b01010,
+		Tim1Cc3			= 0b01011,
+		Tim3Trgo		= 0b01100,
+		ExtiLine3		= 0b01101,
+		Tim6Trgo		= 0b01110,
+		Tim15Trgo		= 0b01111,
+		Tim20Trgo		= 0b10000,
+		Tim20Trgo2		= 0b10001,
+		Tim20Cc2		= 0b10010,
+		HrtimAdcTrg2	= 0b10011,
+		HrtimAdcTrg4	= 0b10100,
+		HrtimAdcTrg5	= 0b10101,
+		HrtimAdcTrg6	= 0b10110,
+		HrtimAdcTrg7	= 0b10111,
+		HrtimAdcTrg8	= 0b11000,
+		HrtimAdcTrg9	= 0b11001,
+		HrtimAdcTrg10	= 0b11010,
+		HrtimAdcTrg1	= 0b11011,
+		HrtimAdcTrg3	= 0b11100,
+		Lptimout		= 0b11101,
+		Tim7Trgo		= 0b11110,
+		//Reserved		= 0b11111,
+%% endif
+%% else
+		ExtTrigger0		= 0b0,
+		ExtTrigger1		= 0b1,
+		ExtTrigger2		= 0b10,
+		ExtTrigger3		= 0b11,
+		ExtTrigger4		= 0b100,
+		ExtTrigger5		= 0b101,
+		ExtTrigger6		= 0b110,
+		ExtTrigger7		= 0b111,
+		ExtTrigger8		= 0b1000,
+		ExtTrigger9		= 0b1001,
+		ExtTrigger10	= 0b1010,
+		ExtTrigger11	= 0b1011,
+		ExtTrigger12	= 0b1100,
+		ExtTrigger13	= 0b1101,
+		ExtTrigger14	= 0b1110,
+		ExtTrigger15	= 0b1111,
+		ExtTrigger16	= 0b10000,
+		ExtTrigger17	= 0b10001,
+		ExtTrigger18	= 0b10010,
+		ExtTrigger19	= 0b10011,
+		ExtTrigger20	= 0b10100,
+		ExtTrigger21	= 0b10101,
+		ExtTrigger22	= 0b10110,
+		ExtTrigger23	= 0b10111,
+		ExtTrigger24	= 0b11000,
+		ExtTrigger25	= 0b11001,
+		ExtTrigger26	= 0b11010,
+		ExtTrigger27	= 0b11011,
+		ExtTrigger28	= 0b11100,
+		ExtTrigger29	= 0b11101,
+		ExtTrigger30	= 0b11110,
+		ExtTrigger31	= 0b11111,
+%% endif
+	};
+
```

Comment:
Data is generated from this script: https://gist.github.com/rleh/3a1669d7c0af5c8aeafa9bdbe7cf1fe9

I also don't really like to put the data in `module.lb`, but since there is already the channel mapping data i will add the trigger lists.
Additionally I will have a look at the STM32F3 triggers in the ref manual.